main.py

```diff
@@ -0,0 +1,32 @@
+# main click program to use GiphyCLI
+import os
+import click
+from giphy_cli import GiphyCLI
+
+@click.group()
+def gif():
+    pass
+
+@gif.command()
+@click.option("--api-key", envvar="GIPHY_API_KEY", required=True, help="Giphy API key")
+@click.option('--limit', default=10, show_default=True, help='Number of trending GIFs to fetch')
+@click.option('--offset', default=0, show_default=True, help='Offset for pagination')
+@click.option('--rating', type=click.Choice(['g', 'pg', 'pg-13', 'r'], case_sensitive=False), default='g', show_default=True, help='Maximum rating of GIFs')
+def trending(api_key, limit, offset, rating):
```

Comment:
Yeah, these are not the correct options for the CLI. Take a look at the project description.